File: Ex78ViewBinding/app/src/main/java/com/gy25m/ex78viewbinding/MyFragment.java
package com.gy25m.ex78viewbinding;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import com.gy25m.ex78viewbinding.databinding.FragmentMyBinding;

public class MyFragment extends Fragment {

    //fragment_my.xml 파일과 연결되어 있는 바인딩 클래스 참조변수
    FragmentMyBinding binding;

    // 이 프래그먼트가 보여줄 뷰를 리턴해주는 기능메소드 재정의

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        binding=FragmentMyBinding.inflate(inflater,container,false);
        return binding.getRoot(); //bind은 뷰가아님
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        binding.btn.setOnClickListener(v -> binding.tv.setText("Good day")); //한줄이면 괄호,; 생략가능
    }
}

File: Ex97DataBinding/app/src/main/java/com/gy25m/ex97databinding/MyFragment.kt
package com.gy25m.ex97databinding

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.databinding.DataBindingUtil
import androidx.fragment.app.Fragment
import com.gy25m.ex97databinding.databinding.FragmentMyBinding

class MyFragment : Fragment() {

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        binding= DataBindingUtil.inflate(inflater, R.layout.fragment_my, container, false)
        return binding.root
    }

    lateinit var binding: FragmentMyBinding

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        //xml 레이아웃에서사용할 데이터클래스 객체 생성 밒 설정
        binding.vm=MyDataViewModel()
    }

}